bert_e/tests/test_bert_e.py

```diff
@@ -450,10 +510,37 @@ def test_branch_cascade_target_hotfix(self):
         fixver = ['6.6.6.3']
         self.finalize_cascade(branches, tags, destination, fixver)
 
-    def test_branch_cascade_invalid_dev_branch(self):
+    def test_branch_cascade_target_three_digit_dev(self):
+        """Test cascade targeting three-digit development branch"""
         destination = 'development/4.3.17'
         branches = OrderedDict({
-            1: {'name': 'development/4.3.17', 'ignore': False}
+            1: {'name': 'development/4.3', 'ignore': True},
+            2: {'name': 'development/4.3.17', 'ignore': False},
+            3: {'name': 'development/5.1', 'ignore': False},
+            4: {'name': 'development/10.0', 'ignore': False}
+        })
+        tags = ['4.3.16', '4.3.17', '5.1.3']
+        fixver = ['4.3.18', '5.1.4', '10.0.0']
+        self.finalize_cascade(branches, tags, destination, fixver)
+
+    def test_branch_cascade_mixed_versions(self):
+        """Test cascade with mix of 2-digit and 3-digit development branches"""
+        destination = 'development/5.1'
+        branches = OrderedDict({
+            1: {'name': 'development/4.3', 'ignore': True},
+            2: {'name': 'development/4.3.17', 'ignore': True},
+            3: {'name': 'development/5.1', 'ignore': False},
+            4: {'name': 'development/5.1.8', 'ignore': False},
+            5: {'name': 'development/10.0', 'ignore': False}
+        })
```

Comment:
Similar story, I expect `5.1.8` to be ignored if `development/5.1` is targetted:

```suggestion
        branches = OrderedDict({
            1: {'name': 'development/4.3.17', 'ignore': True},
            2: {'name': 'development/4.3', 'ignore': True},
            3: {'name': 'development/5.1.8', 'ignore': True},
            4: {'name': 'development/5.1', 'ignore': False},
            5: {'name': 'development/10.0', 'ignore': False}
        })
```

Reply:
Agree